docs/adr/decisions/0011-random-storage-object-ids.md

```diff
@@ -0,0 +1,83 @@
+---
+status: "proposed"
+---
+# Random Storage Object IDs
+
+## Context and Problem Statement
+
+The TAMS `/storage` endpoint provides object IDs and URLs to PUT media objects to the object store.
+The object IDs and URLs are arranged according to the Flow's storage segmentation rate, such that each storage segment (with object ID and URL) is assigned a timerange.
+A client is required to store a media object such that the earliest media timestamp falls within the storage segment's timerange.
+
+Should the TAMS store prescribe how the media is segmented by applications?
+
+## Considered Options
+
+* Option 1: Use the current segmentation strategy
+* Option 2a: Each request returns new object IDs
+* Option 2b: Each request only returns storage segments with media unavailable
+* Option 3: Each request returns new object IDs and doesn't prescribed storage segments
+* Option 4: TAMS provides segmentation hints
+
+## Decision Outcome
+
+Chosen option: Option 3: Return random object IDs and don't prescribed storage segments
+
+It is unclear what hints could be provided to clients that would practically be usefull, and for that reason option 4 is not chosen.
+
+### Implementation
+
+Implemented in [#34](https://github.com/bbc/tams/pull/34).
+
+## Pros and Cons of the Options
+
+### Option 1: Use the current segmentation strategy
+
+The TAMS assigns storage segments using the Flow's storage segmentation rate.
+A storage segment covers a timerange, is assigned a media object ID and provides PUT URL for clients to use.
+A client is required to store a media object such that the earliest media timestamp falls within the storage segment's timerange.
+the timerange.
+
+* Good, because the logic for segmenting media is simple to follow
+* Good, because the media object is overwritten if writing again to a storage segment timerange
```

Comment:
Should we say something like "however clients must take care to ensure the immutability of stored content is respected" or similar here?

Reply:
Agree - updated in https://github.com/bbc/tams/pull/34/commits/3bbb2636a1aca5ea223ccf1ac7e913b140825345